sycl/test-e2e/Matrix/common.hpp

```diff
@@ -127,7 +129,8 @@ void matrix_rand(unsigned int rows, unsigned int cols, T *src, T val) {
 
   for (unsigned int i = 0; i < rows; i++) {
     for (unsigned int j = 0; j < cols; j++) {
-      if constexpr (std::is_same_v<T, bfloat16> || std::is_same_v<T, float>) {
+      if constexpr (std::is_same_v<T, bfloat16> || std::is_same_v<T, float> ||
+                    std::is_same_v<T, double>) {
```

Comment:
question: do we want to use std::uniform_real_distribution<double> for double type?